index.js

```diff
@@ -38,14 +38,26 @@ function fastifyStatic (fastify, opts, next) {
     maxAge: opts.maxAge
   }
 
-  function pumpSendToReply (request, reply, pathname, rootPath) {
+  function pumpSendToReply (request, reply, pathname, rootPath, checkedExtensions) {
     var options = Object.assign({}, sendOptions)
 
     if (rootPath) {
       options.root = rootPath
     }
 
-    const stream = send(request.raw, pathname, options)
+    let encodingExtension = ''
+
+    if (opts.preCompressed) {
+      if (!checkedExtensions) {
+        checkedExtensions = new Set()
```

Comment:
I don't understand this logic. Could you explain?

Reply:
I've got a recursive call to `pumpSendToReply` within the `error` handler and I wanted to avoid re-creating the data structure where we store what we have already checked for on subsequent calls as we attempt to send other potential compressed versions. I thought about using default parameters, but that seemed too eager since this is an opt-in feature. 

Is it possible to have this structure declared at the plugin level or somewhere else?

Reply:
Can you document it a bit? That's ok.